sklearn/preprocessing/_encoders.py

```diff
@@ -731,6 +765,14 @@ def inverse_transform(self, X):
 
         for i in range(n_features):
             labels = X[:, i].astype('int64', copy=False)
-            X_tr[:, i] = self.categories_[i][labels]
+            # set unknown values to None
+            unknown_labels = labels == self.unknown_value
+            if self.handle_unknown == 'use_encoded_value':
+                X_tr[:, i] = np.where(
+                    unknown_labels, None,
+                    self.categories_[i][np.where(
+                        unknown_labels, 0, labels)])
```

Comment:
Hmm do you think the following is clearer:

```py
unknown_labels = labels == self.unknown_value
X_tr[:, i] = self.categories_[i][np.where(unknown_labels, 0, labels)]
X_tr[unknown_labels, i] = None
```

It might be slightly better memory wise as well.

Reply:
Not sure about memory, but yes, your suggestion is clearer.